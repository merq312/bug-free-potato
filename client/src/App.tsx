import { BrowserRouter, Route, Switch } from "react-router-dom"
import HomePage from "./pages/home/Home.page"
import HeaderComponent from "./components/header/Header.component"

function App() {
  return (
    <div className="flex flex-col h-screen">
      <HeaderComponent />
      <main className="grid grid-cols-12 overflow-hidden h-full text-xl sm:text-2xl  bg-gray-700 ">
        <BrowserRouter>
          <Switch>
            <Route exact path="/" component={HomePage} />
          </Switch>
        </BrowserRouter>
      </main>
    </div>
  )
}

export default App
