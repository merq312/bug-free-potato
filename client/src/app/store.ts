import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit"
import { logger } from "redux-logger"
import counterReducer from "../features/counter/counterSlice"
import messageReducer from "../features/messages/messageSlice"
import userReducer from "../features/user/userSlice"
import { createSocketMiddleware } from "../features/middleware/socket.middleware"

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    messages: messageReducer,
    user: userReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(createSocketMiddleware(), logger),
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
