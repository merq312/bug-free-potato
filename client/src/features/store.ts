import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit"
import { logger } from "redux-logger"
import counterReducer from "./counter/counterSlice"
import messageReducer from "./message/messageSlice"
import userReducer from "./user/userSlice"
import roomReducer from "./room/roomSlice"
import { createSocketMiddleware } from "./middleware/socket.middleware"

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    messages: messageReducer,
    user: userReducer,
    rooms: roomReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(createSocketMiddleware(), logger),
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
