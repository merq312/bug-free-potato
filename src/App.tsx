import { Switch, Route, Redirect } from 'react-router-dom'
import Home from './pages/home/home'

function App() {
  return (
    <div className="text-center">
      <header className="min-h-screen flex flex-col items-center justify-center text-2xl">
        <Switch>
          <Route exact path="/" component={Home} />
        </Switch>
      </header>
    </div>
  )
}

export default App
