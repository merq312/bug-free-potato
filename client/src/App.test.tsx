import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import { store } from './features/store'
import App from './App'

describe('Homepage', () => {
  test('App contains main section', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    )

    expect(screen.getByRole('main')).toBeInTheDocument()
  })

  test('App containers header', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    )

    expect(screen.getByRole('banner')).toBeInTheDocument()
  })

  test('renders correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
