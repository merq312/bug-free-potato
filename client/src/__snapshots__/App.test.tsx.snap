// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col h-screen"
  >
    <header
      class="bg-gray-700 flex items-center justify-between w-screen h-16 border-b-2 border-gray-200"
    >
      <h1
        class="text-gray-200 text-xl sm:text-2xl font-bold m-2"
      >
        Some Message App
      </h1>
      <p
        class="text-gray-200 px-4"
      >
        Loading...
      </p>
    </header>
    <main
      class="grid grid-cols-12 overflow-hidden h-full text-xl sm:text-2xl  bg-gray-700 "
    >
      <div
        class="flex flex-col col-start-1 col-end-13 h-full overflow-hidden md:col-start-3 md:col-end-11"
      >
        <div
          class="h-full flex flex-col grid-rows-tabs md:grid md:grid-cols-12 overflow-hidden"
        >
          <div
            class="bg-gray-400 row-span-full md:col-start-1 md:col-end-3"
          />
          <div
            class="flex bg-gray-300 text-gray-800 text-base md:text-xl px-2 py-1 md:col-start-3 md:col-end-13"
          >
            <div
              class="px-4 border-r hover:underline border-gray-700 last:border-r-0 cursor-pointer underline"
            >
              Global
            </div>
          </div>
          <div
            class="scroll flex-grow md:h-auto bg-gray-300 flex flex-col-reverse md:col-start-3 md:col-end-13 overflow-y-scroll"
          >
            <div />
          </div>
        </div>
        <div
          class="bg-gray-300"
        >
          <form
            class="h-full grid md:grid-rows-1 grid-cols-12 items-center justify-around"
          >
            <input
              autocomplete="off"
              class="text-sm md:text-2xl col-start-1 col-end-13 md:col-start-1 md:col-end-3 px-3 text-gray-700 bg-gray-300 focus:outline-none resize-none"
              name="user-name"
              type="text"
              value="Guest"
            />
            <div
              class="col-start-1 md:col-start-3 col-end-13 ml-2 flex"
            >
              <input
                autocomplete="off"
                class="flex-grow min-w-0 my-3 px-3 py-1.5 rounded-xl bg-gray-200 focus:outline-none resize-none"
                name="user-input"
                type="text"
                value=""
              />
              <button
                class="px-2 sm:px-4 text-gray-700 font-bold"
                type="submit"
              >
                Send
              </button>
            </div>
          </form>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
